src/platform/backtrace.cpp

```diff
@@ -21,6 +24,27 @@
 #endif
 
 namespace {
+boost::filesystem::path exeDir()
+{
+    int length = 0, dirname_length = 0;
+    length = wai_getExecutablePath(NULL, 0, &dirname_length);
+    if (length > 0)
+    {
+        std::string path( length + 1, '\0' );
+        wai_getExecutablePath( &path[0], length, &dirname_length);
```

Comment:
This doesn't look perfectly safe: the data for `std::string` it is not guaranteed that to be stored in a single chunk of continuous memory; `std::vector<char>` would be safer.